src/openforms/submissions/api/viewsets.py

```diff
@@ -288,8 +293,18 @@ def update(self, request, *args, **kwargs):
         return Response(serializer.data, status=status_code)
 
     @extend_schema(
-        description=_("Apply/check form logic"),
+        summary=_("Apply/check form logic"),
+        description=_("Apply/check the logic rules specified on the form."),
         request=FormDataSerializer,
+        responses={
+            200: inline_serializer(
+                "SubmissionStateLogicSerializer",
+                {
+                    "submission": SubmissionSerializer(),
+                    "step": SubmissionStepSerializer(),
+                },
+            )
```

Comment:
Please define an explicit serializer for this. Can just be a subclass of `serializers.Serializer` and you pass it a dict as instance, that should work.

Reply:
I couldn't get it to work with the dict as instance, so I added a dataclass instead